PocketKit/Sources/SharedPocketKit/Session/SessionBackupUtility.swift

```diff
@@ -0,0 +1,113 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Combine
+import RNCryptor
+
+/**
+ A utility class that can back up a `SharedPocketKit.Session` to an encrypted legacy Pocket store.
+ This is done by adding / modifying existing key/value pairs in the store.
+ - Note: The utility of this class is that we will want to force-run the legacy migration
+ after transferring orgs, and a Pocket 8-only user won't have a pre-existing (required) store.
+ By backing up a session, we can create this store (or update a pre-existing one) as necessary.
+ */
+public class SessionBackupUtility {
+    /// The decryption key used in conjunction with `store`.
+    private let password: String
+
+    private let userDefaults: UserDefaults
+    private let store: LegacyEncryptedStore
+    private let notificationCenter: NotificationCenter
+
+    private let notificationQueue = DispatchQueue(label: "SessionBackupUtility", qos: .utility)
+    private var subscriptions: Set<AnyCancellable> = []
+
+    public init(userDefaults: UserDefaults, store: LegacyEncryptedStore, notificationCenter: NotificationCenter) {
+        self.userDefaults = userDefaults
+        self.store = store
+        self.notificationCenter = notificationCenter
+
+        // Return an existing decryption key, else create + store a new one, and return that
+        if let existing = userDefaults.string(forKey: LegacyUserMigration.decryptionKey) {
+            self.password = existing
+        } else {
+            // Generate a password the same way as the legacy app
+            let password = RNCryptor.randomData(ofLength: 32).base64EncodedString()
+            userDefaults.set(password, forKey: LegacyUserMigration.decryptionKey)
+            self.password = password
+        }
+    }
+
+    public func start() {
+        // Utilize NotificationCenter to save a user session on-disk after a user has logged in
+        notificationCenter.publisher(for: .userLoggedIn)
+            .receive(on: notificationQueue)
+            .sink { [weak self] notification in
+                guard let session = notification.object as? Session else {
```

Comment:
```suggestion
                guard let session = notification.object as? Session, let self else {
```
simplify the self conditional?